### PR TITLE
EVG-6001: change environment construction

### PR DESCRIPTION
**Pull request by tychoish**
The goal of this patch is mostly to refactor assumptions about the global environment and iron out the tests so they make fewer assumptions about a global environment state. 

This *does* create a connection using the new driver and will use it in the queue reporting UI, but other than that there is no database driver changes in this patch, which is just to set up the changes to the environment so we can make the switch to the new driver in a patch with fewer lines of code changed. 

I think beyond the changes to environment.go and the testutil package the changes don't really have much substance.

### CODE REVIEW

File: environment.go [hunk 1]
```diff
@@ -110,6 +109,53 @@ type Environment interface {
...
+// Configure initializes the object. Some implementations may
+// not allow the same instance to be configured more than
+// once.
+//
+// If Configure returns without an error, you should assume
+// that the queues have been started, there was no issue
+// establishing a connection to the database, and that the
+// local and remote queues have started.
+//
+// Configure requires that either the path or DB is sent so that it
```
Comment (bsamek) [inline environment.go]: nit: s/sent/set on this line and the one below

File: environment.go [hunk 2]
```diff
@@ -110,6 +109,53 @@ type Environment interface {
 	Close(context.Context) error
 }
 
+// Configure initializes the object. Some implementations may
```
Comment (bsamek) [inline environment.go]: nits: s/Configure/NewEnvironment (a few times) and s/object/environment

File: environment.go [hunk 3]
```diff
@@ -110,6 +109,53 @@ type Environment interface {
...
+//
+// If Configure returns without an error, you should assume
+// that the queues have been started, there was no issue
+// establishing a connection to the database, and that the
+// local and remote queues have started.
+//
+// Configure requires that either the path or DB is sent so that it
+// can construct the evergreen settings. If both are sent, the
+// settings will be from the file
+func NewEnvironment(ctx context.Context, confPath string, db *DBSettings) (Environment, error) {
```
Comment (bsamek) [inline environment.go]: This should first validate that either db it not nil or path is non-empty.

File: environment_test.go [hunk 4]
```diff
@@ -60,7 +62,11 @@ func (s *EnvironmentSuite) TestLoadingConfig() {
 	settings, err := NewSettings(s.path)
 	s.NoError(err)
 	db := settings.Database
-	s.NoError(s.env.Configure(ctx, "", &db))
+
+	env, err = NewEnvironment(ctx, "", &db)
+	s.Require().NoError(err)
+	fmt.Println(db, s.env.settings)
```
Comment (bsamek) [inline environment_test.go]: remove print statement

File: environment.go (hunk 3 above)
Comment (tychoish) [reply environment.go]: that's the first statement conditional statement in the current version. 

the fallback behavior here is pretty fiddly/delicate and basically every test fails if we change it a bit.

**Pull request merged**